hawkbit-mgmt-resource/src/main/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtTargetFilterQueryResource.java

```diff
@@ -0,0 +1,180 @@
+        final Slice<TargetFilterQuery> findTargetFiltersAll;
+        final Long countTargetsAll;
+        if (rsqlParam != null) {
+            final Page<TargetFilterQuery> findFilterPage = this.filterManagement
+                    .findTargetFilterQueryByFilter(pageable, rsqlParam);
+            countTargetsAll = findFilterPage.getTotalElements();
+            findTargetFiltersAll = findFilterPage;
+        } else {
+            findTargetFiltersAll = this.filterManagement.findAllTargetFilterQuery(pageable);
+            countTargetsAll = this.filterManagement.countAllTargetFilterQuery();
+        }
+
+        final List<MgmtTargetFilterQuery> rest = MgmtTargetFilterQueryMapper
+                .toResponse(findTargetFiltersAll.getContent());
+        return new ResponseEntity<>(new PagedList<MgmtTargetFilterQuery>(rest, countTargetsAll), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<MgmtTargetFilterQuery> createFilter(@RequestBody MgmtTargetFilterQueryRequestBody filter) {
+        final TargetFilterQuery createdTarget = this.filterManagement
+                .createTargetFilterQuery(MgmtTargetFilterQueryMapper.fromRequest(entityFactory, filter));
+
+        return new ResponseEntity<>(MgmtTargetFilterQueryMapper.toResponse(createdTarget), HttpStatus.CREATED);
+    }
+
+    @Override
+    public ResponseEntity<MgmtTargetFilterQuery> updateFilter(@PathVariable("filterId") Long filterId,
+            @RequestBody MgmtTargetFilterQueryRequestBody targetFilterRest) {
+
+        final TargetFilterQuery existingFilter = findFilterWithExceptionIfNotFound(filterId);
+        LOG.debug("updating target filter query {}", existingFilter.getId());
+        if (targetFilterRest.getName() != null) {
+            existingFilter.setName(targetFilterRest.getName());
+        }
+        if (targetFilterRest.getQuery() != null) {
+            existingFilter.setQuery(targetFilterRest.getQuery());
+        }
+
+        final TargetFilterQuery updateFilter = this.filterManagement.updateTargetFilterQuery(existingFilter);
+
+        return new ResponseEntity<>(MgmtTargetFilterQueryMapper.toResponse(updateFilter), HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<Void> deleteFilter(@PathVariable("filterId") Long filterId) {
+        final TargetFilterQuery filter = findFilterWithExceptionIfNotFound(filterId);
+        this.filterManagement.deleteTargetFilterQuery(filter.getId());
+        LOG.debug("{} target filter query deleted, return status {}", filterId, HttpStatus.OK);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @Override
+    public ResponseEntity<MgmtTargetFilterQuery> postAssignedDistributionSet(@PathVariable("filterId") Long filterId,
+            @RequestBody MgmtId dsId) {
+        final TargetFilterQuery filter = findFilterWithExceptionIfNotFound(filterId);
+
+        DistributionSet distributionSet;
```

Comment:
The method already exist privately in the MgmtDistributionSetResource class. 
Some possible solutions:

1. It could be made public and be defined in the Interface.
2. A new package local utility class is defined that offers Exception throwing methods for all kind of entities.
3. Copy the private method to  MgmtTargetFilterQueryResource
4. Leave it as is, because its only one case in the MgmtTargetFilterQueryResource class.